Userland/Libraries/LibWeb/HTML/ShadowRealmGlobalScope.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024, Shannon Booth <shannon@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/ShadowRealmGlobalScopePrototype.h>
+#include <LibWeb/HTML/ShadowRealmGlobalScope.h>
+
+namespace Web::HTML {
+
+JS_DEFINE_ALLOCATOR(ShadowRealmGlobalScope);
+
+ShadowRealmGlobalScope::ShadowRealmGlobalScope(JS::Realm& realm)
+    : DOM::EventTarget(realm)
+{
+}
+
+ShadowRealmGlobalScope::~ShadowRealmGlobalScope() = default;
+
+JS::NonnullGCPtr<ShadowRealmGlobalScope> ShadowRealmGlobalScope::create(JS::Realm& realm)
+{
+    return realm.heap().allocate<ShadowRealmGlobalScope>(realm, realm);
+}
+
+void ShadowRealmGlobalScope::initialize(JS::Realm&)
+{
+    // FIXME: This does not work as we do not have any intrinsics in the [HostDefined] of a shadow realm. Figure out how this _should_ work.
+    // Base::initialize(realm);
+    // WEB_SET_PROTOTYPE_FOR_INTERFACE(ShadowRealmGlobalScope);
```

Comment:
each [Global] should have a `add_scopename_exposed_interfaces`, as well as a call to initialize its `ScopeNameGlobalScopeGlobalMixin::initialize(realm, *this)`

In the case of the other globals, this is called from their ESO's `setup` method.

From WorkerEnvironmentSettingsObject::setup()

```
    // 8. Set realm's [[HostDefined]] field to settings object.
    auto intrinsics = realm->heap().allocate<Bindings::Intrinsics>(*realm, *realm);
    auto host_defined = make<Bindings::HostDefined>(settings_object, intrinsics, page);
    realm->set_host_defined(move(host_defined));

    // Non-Standard: We cannot fully initialize worker object until *after* the we set up
    //    the realm's [[HostDefined]] internal slot as the internal slot contains the web platform intrinsics
    worker.initialize_web_interfaces({});
```

This ends up calling WorkerGlobalScope::intialize_web_interfaces (custom) which calls into DedicatedWorkerGlobalScope::initialize_web_interfaces_impl that does the calling of those two methods.